lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -253,19 +251,35 @@ abstract class TextEditingStrategy {
   void setEditingState(EditingState editingState);
 
   /// Set style to the native DOM element used for text editing.
-  void updateElementStyle(_EditingStyle style);
+  void updateElementStyle(EditableTextStyle style);
 
   /// Disables the element so it's no longer used for text editing.
   ///
   /// Calling [disable] also removes any registered event listeners.
   void disable();
 }
 
+/// A [TextEditingStrategy] that places its [domElement] assuming no
+/// prior transform or sizing is applied to it.
+///
+/// This implementation is used by text editables when semantics is not
+/// enabled. With semantics enabled the placement is provided by the semantics
+/// tree.
+class GloballyPositionedTextEditingStrategy extends DefaultTextEditingStrategy {
+  GloballyPositionedTextEditingStrategy(HybridTextEditing owner) : super(owner);
```

Comment:
now we are adding one more intermediate node to the class diagram.

I am curious why we did not just override placeElement (used to called positionlement) and remove @mustcall annotation from there.

Can you explain the reasoning preferably in the documentation so that next time another functionality comes we decide between creating one more intermediate node or just overriding the method?

Reply:
This class supplies functionality (global placement) that's relevant to its subclasses and not to semantics, so I separated it out. There's no other reason. If we move this logic into `DefaultTextEditingStrategy` then the semantics implementation _will_ depend on it but will have to undo it.

I prefer this approach because I find it easier to understand code that only adds missing functionality than understanding code the alters or undoes existing functionality.

For the purposes of this PR I'm happy either way, so LMK what you prefer.

Reply:
I think these already exist in the current code.

I think it is important to have a good class design and rely on solid patterns. When to do that mostly should depend on the project priorities. I believe the need purify this class has not emerged. Yet since you already wrote this code, let's get this one in.